src/Skoruba.IdentityServer4.Admin.EntityFramework.Identity/Repositories/IdentityRepository.cs

```diff
@@ -271,26 +271,34 @@ public virtual async Task<PagedList<TRoleClaim>> GetRoleClaimsAsync(string roleI
             pagedList.PageSize = pageSize;
 
             return pagedList;
-        }
-
-		public virtual async Task<PagedList<TRoleClaim>> GetUserRoleClaimsAsync(string userId, string claimSearchText, int page = 1, int pageSize = 10)
-		{
-			var id = ConvertUserKeyFromString(userId);
-			Expression<Func<TRoleClaim, bool>> searchCondition = x => x.ClaimType.Contains(claimSearchText);
-			var claimsQ = DbContext.Set<TUserRole>().Where(x => x.UserId.Equals(id))
-				.Join(DbContext.Set<TRoleClaim>().WhereIf(!string.IsNullOrEmpty(claimSearchText), searchCondition), ur => ur.RoleId, rc => rc.RoleId, (ur, rc) => rc);
-
-			var claims= await claimsQ.PageBy(x => x.Id, page, pageSize)
-				.ToListAsync();
-
-			var pagedList = new PagedList<TRoleClaim>();
-			pagedList.Data.AddRange(claims);
-			pagedList.TotalCount = await claimsQ.CountAsync();
-			pagedList.PageSize = pageSize;
-
-			return pagedList;
-		}
-
+        }
+
+		public virtual async Task<PagedList<TRoleClaim>> GetUserRoleClaimsAsync(string userId, string claimSearchText, int page = 1, int pageSize = 10)
+		{
+			var id = ConvertUserKeyFromString(userId);
+			Expression<Func<TRoleClaim, bool>> searchCondition = x => x.ClaimType.Contains(claimSearchText);
+			var claimsQ = DbContext.Set<TUserRole>().Where(x => x.UserId.Equals(id))
+				.Join(DbContext.Set<TRoleClaim>().WhereIf(!string.IsNullOrEmpty(claimSearchText), searchCondition), ur => ur.RoleId, rc => rc.RoleId, (ur, rc) => rc);
+
+			var claims= await claimsQ.PageBy(x => x.Id, page, pageSize)
+				.ToListAsync();
+
+			var pagedList = new PagedList<TRoleClaim>();
+			pagedList.Data.AddRange(claims);
+			pagedList.TotalCount = await claimsQ.CountAsync();
+			pagedList.PageSize = pageSize;
+
+			return pagedList;
+		}
+
+		public virtual Task<TUserClaim> GetUserClaimByTypeAsync(string userId, string claimType)
```

Comment:
This new function allows the ability to search for a user claim by "type" text instead of by known ID.  For the invite system this would be searching for the existence of the "invitation_token" claim.